compiler/rustc_middle/src/ty/context.rs

```diff
@@ -2104,6 +2106,44 @@ impl<'tcx> TyCtxt<'tcx> {
         lint_level(self.sess, lint, level, src, Some(span.into()), msg, decorate);
     }
 
+    /// Find the crate root and the appropriate span where `use` and outer attributes can be
+    /// inserted at.
+    pub fn crate_inject_span(self, hir_id: HirId) -> Option<Span> {
```

Comment:
The name is not very clear to me... maybe something like `crate_level_attribute_injection_span`?

Cause the current version I read as "inject a span into a crate"...